src/allmydata/storage/server.py

```diff
@@ -777,7 +778,7 @@ def advise_corrupt_share(self, share_type, storage_index, shnum,
             si_s,
             shnum,
         )
-        with open(report_path, "w") as f:
+        with open(report_path, "w", encoding="utf-8") as f:
```

Comment:
I think that `open(report_path, "w")` opened a text-mode file and that `report` was unicode (on Python 3) before and is still unicode - so the `encoding="utf-8"` doesn't *usually* change any behavior but maybe it prevents Python from picking a weird alternate encoding in some runtime environments?

Reply:
On Windows default was cp1252 on the Windows CI builders, at least. So that's the behavior change.

(The motivation is that if we're sending Unicode reasons, it's not always encodable in cp1252 or other limited character sets).